Edited README (#978)
themoeway/yomitan: README.md

- 📈 Interactive popup definition window for displaying search results.
- 🔊 Built-in audio playback for select Japanese words.
- ✍ Kanji stroke order diagrams are generally just a click away.
- 📝 [Automatic note creation](./docs/anki-integration.md#anki-integration) for the [Anki](https://apps.ankiweb.net/) flashcard program via the [AnkiConnect](https://foosoft.net/projects/anki-connect) plugin.
- 🔍 Custom search page for easily executing custom search queries.
- 📖 Support for multiple dictionary formats including [EPWING](https://ja.wikipedia.org/wiki/EPWING) via the [Yomitan Import](https://github.com/themoeway/yomitan-import) tool.
- ✨ Clean, modern code makes it easy for developers to [contribute](#contributing) new features.

[![Term definitions](img/ss-terms-thumb.png)](img/ss-terms.png)
[![Kanji information](img/ss-kanji-thumb.png)](img/ss-kanji.png)
[![Dictionary options](img/ss-dictionaries-thumb.png)](img/ss-dictionaries.png)
[![Anki options](img/ss-anki-thumb.png)](img/ss-anki.png)

## Helpful information<!-- omit from toc -->
## Helpful information

> - [Migrating from Yomichan (legacy)](./docs/yomichan-migration.md#migrating-from-yomichan)

Reviewer: Why remove?

Author: I moved it to the top

Reviewer: ![2024-05-19_18-54-13](https://github.com/themoeway/yomitan/assets/17340496/50914f77-0079-4a12-af15-7a88169d3e98)

Dictionaries seem to be gone, and I think it's useful to have the links clearly visible as a toc like that.

Author: :thumb:

Reviewer: Can you restore the migration guide link as well? I'm hesitant to remove stuff from there since it's where you'd usually go to look for docs, and it's a link to a separate page.

Author: 👍

[![OpenSSF Scorecard](https://api.securityscorecards.dev/projects/github.com/themoeway/yomitan/badge)](https://securityscorecards.dev/viewer/?uri=github.com/themoeway/yomitan)
[![Discord](https://dcbadge.vercel.app/api/server/UGNPMDE7zC?style=flat)](https://discord.gg/UGNPMDE7zC)

:wave: **Yomitan is [the successor](https://foosoft.net/posts/passing-the-torch-to-yomitan/) to Yomichan** ([migration guide](./docs/yomichan-migration.md#migrating-from-yomichan)) which was [sunset](https://foosoft.net/posts/sunsetting-the-yomichan-project/) by its owner on Feb 26, 2023. We have made a number of [foundational changes](#changes) to ensure **the project stays alive, works on latest browser versions, and is easy to contribute to**.

📢 **New contributors [welcome](#contributing)!**

📢 **Interested in adding a new language to Yomitan? See [here](./docs/development/language-features.md) for thorough documentation!**

## What is Yomitan?

Yomitan turns your web browser into a tool for building Japanese language literacy by helping you **read** texts that would otherwise be too difficult to tackle.

> Yomitan provides advanced features not available in other browser-based dictionaries:

Reviewer 2: ```suggestion
Yomitan provides powerful features not available in other browser-based dictionaries:
```

- 💬 Interactive popup definition window for displaying search results.
- 🔊 Built-in native audio sources for Japanese words with the ability to add your own [custom audio sources](./docs/audio.md).
- ✍️ Kanji stroke order diagrams are just a click away.

> - 📝 [Automatic note creation](./docs/anki-integration.md#anki-integration) for the [Anki](https://apps.ankiweb.net/) flashcard program via the [AnkiConnect](https://foosoft.net/projects/anki-connect) plugin.

Reviewer 2: ```suggestion
- 📝 [Automatic flashcard creation](./docs/anki-integration.md#anki-integration) for the [Anki](https://apps.ankiweb.net/) flashcard program via the [AnkiConnect](https://foosoft.net/projects/anki-connect) plugin.
```

Reviewer 3: Should say "card" not flashcard imo since "flashcard program" is right after it.

- 🔍 Custom search page for easily executing custom search queries.
- 📖 Support for multiple dictionary formats including [EPWING](https://ja.wikipedia.org/wiki/EPWING) via the [Yomitan Import](https://github.com/themoeway/yomitan-import) tool.

> - ✨ Clean, modern code makes it easy for developers to [contribute](#contributing) new features.

Reviewer 2: ```suggestion
- ✨ Clean, modern code makes it easy for developers to [contribute](#contributing) new features and languages.
```

# Yomitan

[![Chrome Release](<https://img.shields.io/chrome-web-store/v/likgccmbimhjbgkjambclfkhldnlhbnn?logo=Google%20Chrome&logoColor=lightblue&color=lightblue&label=chrome%20release%20(stable)>)](https://chrome.google.com/webstore/detail/yomitan/likgccmbimhjbgkjambclfkhldnlhbnn)
[![Firefox Release](<https://img.shields.io/amo/v/yomitan?logo=Firefox&color=orange&label=firefox%20release%20(stable)>)](https://addons.mozilla.org/en-US/firefox/addon/yomitan/)
[![OpenSSF Scorecard](https://api.securityscorecards.dev/projects/github.com/themoeway/yomitan/badge)](https://securityscorecards.dev/viewer/?uri=github.com/themoeway/yomitan)
[![Discord](https://dcbadge.vercel.app/api/server/UGNPMDE7zC?style=flat)](https://discord.gg/UGNPMDE7zC)

> :wave: **This project is a community fork of Yomichan** ([migration guide](./docs/yomichan-migration.md#migrating-from-yomichan)) which was [sunset](https://foosoft.net/posts/sunsetting-the-yomichan-project/) by its owner on Feb 26 2023. We have made a number of [foundational changes](#changes) to ensure **the project stays alive, works on latest browser versions, and is easy to contribute to**.

Reviewer 3: ```suggestion
:wave: **This project is a community fork of Yomichan** ([migration guide](./docs/yomichan-migration.md#migrating-from-yomichan)) which was [sunset](https://foosoft.net/posts/sunsetting-the-yomichan-project/) by its owner on Feb 26, 2023. We have made a number of [foundational changes](#changes) to ensure **the project stays alive, works on latest browser versions, and is easy to contribute to**.
```

> :wave: **This project is a community fork of Yomichan** ([migration guide](./docs/yomichan-migration.md#migrating-from-yomichan)) which was [sunset](https://foosoft.net/posts/sunsetting-the-yomichan-project/) by its owner on Feb 26, 2023. We have made a number of [foundational changes](#changes) to ensure **the project stays alive, works on latest browser versions, and is easy to contribute to**.

Reviewer 4: Maybe we can now be more upfront about being the "definitive successor to Yomichan".

Reviewer 3: Yeah can link this too to show foosoft's support of yomitan https://foosoft.net/posts/passing-the-torch-to-yomitan/

Author: changed

> 📢 **New contributers 👨‍💻 [welcome](#contributing)!**

Reviewer 3: 👨‍💻 is a horrible emoji and as soon as I see someone use it my opinion of their project's seriousness lowers slightly.
```suggestion
📢 **New contributers [welcome](#contributing)!**
```

> 📢 **Seeking language experts to [add more languages](./docs/development/language-features.md) to Yomitan!**

Reviewer 4: There is no need to be an expert, or even be proficient at a language, to contribute to language features. I've tried to set the bar as low as possible in the docs and such.

> 📢 **Seeking language experts 🎓 to [add more languages](./docs/development/language-features.md) to Yomitan!**

Reviewer 3: Not a fan of the 🎓 breaking up the sentence here.
```suggestion
📢 **Seeking language experts to [add more languages](./docs/development/language-features.md) to Yomitan!**
```

## What is Yomitan?

> Yomitan turns your web browser into a tool for building Japanese language literacy by helping you to **read** texts which would otherwise be too difficult to tackle.

Reviewer 3: ```suggestion
Yomitan turns your web browser into a tool for building Japanese language literacy by helping you **read** texts that would otherwise be too difficult to tackle.
```

> This extension is similar to [10ten Japanese Reader (formerly Rikaichamp)](https://addons.mozilla.org/en-US/firefox/addon/10ten-ja-reader/) for Firefox and [Rikaikun](https://chrome.google.com/webstore/detail/rikaikun/jipdnfibhldikgcjhfnomkfpcebammhp) for Chrome.

Reviewer 4: I think this line is very old, not sure if these still have name recognition.

Author: removed

Yomitan provides advanced features not available in other browser-based dictionaries:

> - 📈 Interactive popup definition window for displaying search results.

Reviewer 3: Popups are more like message boxes than graphs.
```suggestion
- 💬 Interactive popup definition window for displaying search results.
```

> - 🔊 Built-in audio playback for select Japanese words.

Reviewer 3: Should this really say "Japanese words"? Do we not have audio support for any other languages yet?

Author: only defaults are japanese and tts, ill make it more clear

> - ✍ Kanji stroke order diagrams are generally just a click away.

Reviewer 3: Not sure why the base unicode writing hand was used instead of the emoji version.
```suggestion
- ✍️ Kanji stroke order diagrams are generally just a click away.
```

- 📝 [Automatic note creation](./docs/anki-integration.md#anki-integration) for the [Anki](https://apps.ankiweb.net/) flashcard program via the [AnkiConnect](https://foosoft.net/projects/anki-connect) plugin.
- 🔍 Custom search page for easily executing custom search queries.
- 📖 Support for multiple dictionary formats including [EPWING](https://ja.wikipedia.org/wiki/EPWING) via the [Yomitan Import](https://github.com/themoeway/yomitan-import) tool.
- ✨ Clean, modern code makes it easy for developers to [contribute](#contributing) new features.

[![Term definitions](img/ss-terms-thumb.png)](img/ss-terms.png)
[![Kanji information](img/ss-kanji-thumb.png)](img/ss-kanji.png)
[![Dictionary options](img/ss-dictionaries-thumb.png)](img/ss-dictionaries.png)
[![Anki options](img/ss-anki-thumb.png)](img/ss-anki.png)

## Helpful information

- [Migrating from Yomichan (legacy)](./docs/yomichan-migration.md#migrating-from-yomichan)
- [Frequently asked questions](./docs/faq.md#frequently-asked-questions)
- [Recommended and custom dictionary import guide](./docs/dictionaries.md#dictionaries)

> - [Anki integration and flashcards creation guide](./docs/anki-integration.md#anki-integration)

Reviewer 3: ```suggestion
- [Anki integration and flashcard creation guide](./docs/anki-integration.md#anki-integration)
```

- [Keyboard shortcuts](./docs/keyboard-shortcuts.md)
- [Advanced options, including MeCab](./docs/advanced-options.md#advanced-options)

## Installation

> Yomitan comes in two flavors: _stable_ and _testing_. New changes are initially introduced into the _testing_ version, and after some time spent ensuring that they are relatively bug free, they will be promoted to the _stable_ version. If you are technically savvy and don't mind [submitting issues](#contributing) on GitHub, try the _testing_ version; otherwise, the _stable_ version will be your best bet.

Reviewer 3: ```suggestion
Yomitan comes in two flavors: _stable_ and _testing_. New changes are initially introduced into the _testing_ version, and after some time spent ensuring that they are relatively bug free, they will be promoted to the _stable_ version. If you are technically savvy and don't mind [submitting issues](https://github.com/themoeway/yomitan/issues/new/choose) on GitHub, try the _testing_ version; otherwise, the _stable_ version will be your best bet. Check [contributing](#contributing) for more information on how to help.
```

5.  Click on individual kanji in the term definition results to view additional information about those characters, including stroke order diagrams, readings, meanings, and other useful data.

    <img src="resources/images/search-popup-kanji.png" width="300" alt="popup with kanji details">

To further enhance your Yomitan experience, it's worth [integrating with Anki](./docs/anki-integration.md), a spaced-repetition flashcard program to help solidify the words you encounter.

## Changes

> While we have made some substantial changes, the majority of the extension's functionality is thanks to hard work of [FooSoft](https://foosoft.net/) and many other incredible open source contributors from 2016-2023.

Major changes:

- 💪 Completed the Manifest V2 → V3 transition, [read why here](https://developer.chrome.com/blog/resuming-the-transition-to-mv3/)!
- ⚡ Switched to using ECMAScript modules and npm-sourced dependencies.
- 🔧 Implemented an end-to-end CI/CD pipeline.
- 🧪 Switched to standard testing frameworks, vitest and playwrights.

> In addition, we have made important bug fixes and minor enhancements:

Reviewer 4: Should mention supporting other languages here I think.

Author: done

To further enhance your Yomitan experience, it's worth [integrating with Anki](./docs/anki-integration.md), a spaced-repetition flashcard program to help solidify the words you encounter.

## Changes

> While we have made some substantial changes, the majority of the extension's functionality is thanks to hard work of [FooSoft](https://foosoft.net/) and many other incredible open source contributors from 2016-2023.

Major changes:

> - Completed the Manifest V2 → V3 transition, [read why here](https://developer.chrome.com/blog/resuming-the-transition-to-mv3/)!

Reviewer: Good spot to add more emojis. Like 💪 🧪 🔧 ⚡ 📚 ✨

- Switched to using ECMAScript modules and npm-sourced dependencies.
- Implemented an end-to-end CI/CD pipeline.
- Switched to standard testing frameworks, vitest and playwrights.

In addition, we have made important bug fixes and minor enhancements:

- Improved dictionary import speeds by 2x~10x.
- Added functionality to import/export multiple dictionaries, enabling portability across devices.
- And [more](https://github.com/themoeway/yomitan/pulls?q=is%3Apr+is%3Amerged+-label%3Aarea%2Fdependencies+-label%3Akind%2Fmeta).

Since the owner requested forks be uniquely named, we have chosen a new name, _Yomitan_ (_-tan_ is an honorific used for anthropomorphic moe characters).

## Contributing

🚀 **Dip your toes into contributing by looking at issues with the label [good-first-issue](https://github.com/themoeway/yomitan/issues?q=is%3Aopen+is%3Aissue+label%3Agood-first-issue).**

Since this is a distributed effort, we **highly welcome new contributors**! Feel free to browse the [issue tracker](https://github.com/themoeway/yomitan/issues), and read our [contributing guidelines](./CONTRIBUTING.md).

Here are some ways anyone can help:

> * Try using the Yomitan dev build. Not only do you get the latest features, you can help uncover bugs and give feedback to developers early on.

Reviewer: The reason the [first test failed](https://github.com/themoeway/yomitan/actions/runs/9152448266/job/25159912002?pr=978#step:6:69) is because it doesn't adhere to the style guidelines. For reference, here's what happens when you run prettier on this PR

![Code_README md_(Working_Tree)_(README md)_-_yomitan_-_V_2024-05-19_19-19-23](https://github.com/themoeway/yomitan/assets/17340496/3f662f6a-e3b2-4ee0-81ca-4ac189a640c3)

Author: Thank you for the help!

To further enhance your Yomitan experience, it's worth [integrating with Anki](./docs/anki-integration.md), a spaced-repetition flashcard program to help solidify the words you encounter.

## Changes

> While we have made some substantial changes, the majority of the extension's functionality is thanks to hard work of [FooSoft](https://foosoft.net/) and many other incredible open source contributors from 2016-2023.

Major changes:

- 💪 Completed the Manifest V2 → V3 transition, [read why here](https://developer.chrome.com/blog/resuming-the-transition-to-mv3/)!
- ⚡ Switched to using ECMAScript modules and npm-sourced dependencies.
- 🔧 Implemented an end-to-end CI/CD pipeline.
- 🧪 Switched to standard testing frameworks, vitest and playwrights.

In addition, we have made important bug fixes and minor enhancements:

> - ✨ Improved dictionary import speeds by 2x~10x.
> - 📚 Added functionality to import/export multiple dictionaries, enabling portability across devices.
> - 🕵️‍♂ And [more](https://github.com/themoeway/yomitan/pulls?q=is%3Apr+is%3Amerged+-label%3Aarea%2Fdependencies+-label%3Akind%2Fmeta).

Reviewer 3: I don't think the detective emoji fits for this last bit and I think it would be better to use a book for the dictionary.
```suggestion
- 📖 Improved dictionary import speeds by 2x~10x.
- 📚 Added functionality to import/export multiple dictionaries, enabling portability across devices.
- ✨ And [more](https://github.com/themoeway/yomitan/pulls?q=is%3Apr+is%3Amerged+-label%3Aarea%2Fdependencies+-label%3Akind%2Fmeta).
```

Since the owner requested forks be uniquely named, we have chosen a new name, _Yomitan_ (_-tan_ is an honorific used for anthropomorphic moe characters).

## Contributing

🚀 **Dip your toes into contributing by looking at issues with the label [good-first-issue](https://github.com/themoeway/yomitan/issues?q=is%3Aopen+is%3Aissue+label%3Agood-first-issue).**

Since this is a distributed effort, we **highly welcome new contributors**! Feel free to browse the [issue tracker](https://github.com/themoeway/yomitan/issues), and read our [contributing guidelines](./CONTRIBUTING.md).

Here are some ways anyone can help:

> - Try using the Yomitan dev build. Not only do you get the latest features, you can help uncover bugs and give feedback to developers early on.

Reviewer 3: ```suggestion
- Try using the Yomitan dev build. Not only do you get cutting edge features, but you can help uncover bugs and give feedback to developers early on.
```

- Document any UI/UX friction in Github Issues. We're looking to make Yomitan more accessible to non-technical users.

> - All the issues in `area/bug` older than 2 months need help reproducing. If anything interests you, please try to reproduce it and report your results. We can't easily tell if these issues are one-off, have since resolved, or are no longer relevant.

Reviewer 3: ```suggestion
- All the issues in `area/bug` older than 2 months need help reproducing. If anything interests you, please try to reproduce it and report your results. We can't easily tell if these issues are one-off, have since been resolved, or are no longer relevant.
```

> The current active maintainers of Yomitan spend a lot of their time debugging and triaging issues. When someone files a bug report, we need to assess the frequency and severity of the bug. It is extremely helpful if we get multiple reports of people who experience a bug or people who can contribute additional detail to an existing bug report.

If you're looking to code, please let us know what you plan on working on before submitting a Pull Request. This gives the core maintainers an opportunity to provide feedback early on before you dive too deep. You can do this by opening a Github Issue with the proposal.

Some contributions we always appreciate:

> - Well-written tests covering different functionalities. This includes [playwright tests](https://github.com/themoeway/yomitan/tree/master/test/playwright), [benchmark tests](https://github.com/themoeway/yomitan/tree/master/benches), and unit tests
> - Increasing our type coverage
> - More and better documentation!

Reviewer 3: ```suggestion
- Well-written tests covering different functionalities. This includes [playwright tests](https://github.com/themoeway/yomitan/tree/master/test/playwright), [benchmark tests](https://github.com/themoeway/yomitan/tree/master/benches), and unit tests.
- Increasing our type coverage.
- More and better documentation!
```

Information on how to setup and build the codebase can be found [here](./CONTRIBUTING.md#setup).

If you want to add or improve support for a language, read the documentation on [language features](./docs/development/language-features.md).

> Feel free to join us on [TheMoeWay Discord 🐱](https://discord.gg/UGNPMDE7zC) at [#yomitan-development](https://discord.com/channels/617136488840429598/1081538711742844980).

Reviewer 3: ```suggestion
Feel free to join us on [TheMoeWay Discord 🐇](https://discord.gg/UGNPMDE7zC) at [#yomitan-development](https://discord.com/channels/617136488840429598/1081538711742844980).
```
